### PR TITLE
Unit test coverage based on matter cert test case audit

### PR DESCRIPTION
**Pull request by cjswedes**
* Make mock devices more like real devices
* Write new unit tests to improve code coverage
* Note unsupported functionality in driver code comments

- [x] matter sensor driver audit
- [x] matter window covering driver audit
- [x] matter switch driver audit
- [x] matter thermostat driver audit
- [x] Matter lock driver audit
- [x] Matter media driver audit

### CODE REVIEW

File: drivers/SmartThings/matter-media/src/init.lua [hunk 1]
```diff
@@ -207,6 +212,7 @@ local matter_driver_template = {
       },
       [clusters.MediaPlayback.ID] = {
         [clusters.MediaPlayback.attributes.CurrentState.ID] = media_playback_state_attr_handler,
+        --TODO [clusters.MediaPlayback.attributes.PlaybackSpeed.ID] = media_playback_speed_attr_handler,
```
Comment (cjswedes) [inline drivers/SmartThings/matter-media/src/init.lua:215]: I intend to add this functionality in this PR

File: drivers/SmartThings/matter-media/src/init.lua [hunk 2]
```diff
@@ -101,8 +106,7 @@ local function media_playback_state_attr_handler(driver, device, ib, response)
     [CurrentState.PLAYING] = attr.playing(),
     [CurrentState.PAUSED] = attr.paused(),
     [CurrentState.NOT_PLAYING] = attr.stopped(),
-    -- TODO: Update to use buffering capability attribute once it is available
-    [CurrentState.BUFFERING] = attr.playing()
+    [CurrentState.BUFFERING] = attr.buffering()
```
Comment (cjswedes) [inline drivers/SmartThings/matter-media/src/init.lua:109]: @ctowns is it okay to do this? I see the capability changes were done 7 months ago, so I think it should be present in all hubs that have 45.

File: drivers/SmartThings/matter-media/src/init.lua (hunk 1 above)
Comment (cjswedes) [reply drivers/SmartThings/matter-media/src/init.lua]: Opted for just calling out that Advanced Seek features are not supported. It ended up being quite complicated.

**Comment by cjswedes** [comment]
> Run actions/github-script@v3
  with:
    script: core.setFailed('The driver failed to upload: ErrorResponse{ error=Error{ code=TooManyRequestError, target=null, message='null', details=[Error{ code=TooManyRequestError, target=null, message='attempt 1[2](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/actions/runs/4620434679/jobs/8170542880#step:4:2) limit 10 - retry in 5 millis', details=[] }] } }')

The driver upload is failing because of TooManyRequestError. I think this is caused by the package [requests](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/actions/runs/4620434679/workflow#L100). It doesn't seem to affect PRs with fewer drivers changed, and I dont think it will affect normal driver deploys.

**Comment by greens** [comment]
@cjswedes yeah, known issue with rate limiting

**Review by greens — APPROVED** [review]

**Pull request merged**